Source/WebCore/rendering/style/PositionArea.h

```diff
@@ -0,0 +1,113 @@
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+namespace WebCore {
+
+enum CSSValueID : uint16_t;
+
+class CSSValue;
+
+/*
+(span-)left        -> horizontal, start, span?
+(span-)right       -> horizontal, end, span?
+(span-)top         -> vertical, start, span?
+(span-)bottom      -> vertical, end, span?
+
+(span-)x(self-)-start, -end     -> X, start/end, span?, self?
+(span-)y(self-)-start, -end     -> Y, start/end, span?, self?
+
+(span-)(self-)block-start, -end -> block, start/end, span?, self?
+(span-)(self-)inline-start, -end -> inline, start/end, span?, self?
+
+(span-)(self-)start, -end -> (block or inline depending on position in pair), start/end, span?, self?
+
+center      -> (opposite position of other in pair. If other is ambiguous, block), center
+span-all    -> (opposite position of other in pair. If other is ambiguous, block), span-all (or center + span ??)
+*/
+
+// 6 values, 3 bits
+enum class PositionAreaAxis {
+    Horizontal,
+    Vertical,
+    X,
+    Y,
+    Block,
+    Inline,
+};
+
+// 6 values, 3 bits
+enum class PositionAreaTrack {
+    Start,
+    SpanStart,
+    End,
+    SpanEnd,
+    Center,
+    SpanAll,
+};
+
+enum class PositionAreaSelf { No, Yes };
+
+class PositionAreaSpan {
+public:
+    PositionAreaSpan(PositionAreaAxis, PositionAreaTrack, PositionAreaSelf);
+
+    PositionAreaAxis axis() const { return m_axis; }
+    PositionAreaTrack track() const { return m_track; }
+    PositionAreaSelf self() const { return m_self; }
+
+    CSSValueID toKeyword() const;
+
+    friend bool operator==(PositionAreaSpan, PositionAreaSpan) = default;
+
+private:
+    PositionAreaAxis m_axis;
+    PositionAreaTrack m_track;
+    PositionAreaSelf m_self;
+};
+
+WTF::TextStream& operator<<(WTF::TextStream&, const PositionAreaSpan&);
+
+class PositionArea {
+public:
+    PositionArea(PositionAreaSpan span1, PositionAreaSpan span2);
+
+    static PositionArea fromKeyword(CSSValueID);
+    static PositionArea fromKeywordPair(CSSValueID kw1, CSSValueID kw2);
+
+    PositionAreaSpan span1() const { return m_span1; }
+    PositionAreaSpan span2() const { return m_span2; }
```

Comment:
This can also be named accordingly, e.g. `blockOrXAxis()` and `inlineOrYAxis()`.